core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -129,6 +129,21 @@ private[spark] object UI {
     .intConf
     .createWithDefault(1000)
 
+  val UI_TIMELINE_JOBS_MAXIMUM = ConfigBuilder("spark.ui.timeline.jobs.maximum")
```

Comment:
Instead of introducing the new config can we try using these config
`val MAX_RETAINED_JOBS = ConfigBuilder("spark.ui.retainedJobs") .version("1.2.0") .intConf .createWithDefault(1000)`

same is for the stages(spark.ui.retainedStages)

Reply:
How many  records for jobs information that user want them to be retained is not always the same as how many items displayed in the timeline view.
Users can want much more records to be retained than the default number  but want less items to be shown.
So I prefer to have a new property for jobs. The same applies to stages.